pallets/communities/src/functions/registry.rs

```diff
@@ -0,0 +1,48 @@
+use super::*;
+
+impl<T: Config> Pallet<T> {
+	pub(crate) fn community_exists(community_id: &CommunityIdOf<T>) -> bool {
+		Self::community(community_id).is_some()
+	}
+
+	/// Stores an initial info about the community
+	/// Sets the caller as the community admin, the initial community state
+	/// to its default value(awaiting)
+	pub(crate) fn do_register_community(who: &AccountIdOf<T>, community_id: &CommunityIdOf<T>) -> DispatchResult {
+		// Check that the community doesn't exist
+		if Self::community_exists(community_id) {
+			return Err(Error::<T>::CommunityAlreadyExists.into());
+		}
+
+		<CommunityInfo<T>>::insert(
+			community_id.clone(),
+			Community {
+				admin: who.clone(),
+				state: Default::default(),
+				sufficient_asset_id: None,
+			},
+		);
+
+		Self::do_insert_member(community_id, who)?;
+
+		Ok(())
+	}
+
+	pub(crate) fn do_set_metadata(
+		community_id: &CommunityIdOf<T>,
+		value: types::CommunityMetadata<T>,
+	) -> DispatchResult {
+		<pallet::CommunityMetadata<T>>::try_mutate(community_id, |metadata| {
+			if let Some(metadata) = metadata {
```

Comment:
Both branches of the if-let-else seem to do pretty much the same, setting every field separately here is not much different from just setting the entire struct.

Reply:
Here I tried to make an insert/append kind of behaviour. Which would you think would be a better way to achieve it?

Reply:
Just replace the metadata like you have it in the else block `*metadata = Some(value);` assigning all the fields of `value` to the respective fields of `metadata` is semantically the same as replacing entire struct ... with extra steps. It would be worth it if the fields of `value` were optional or needed some validation of assigning some defaults but plainly assigning all the fields has no benefit over just doing what's in line 42

Reply:
Gotcha! Considering the overriding actually occurs on the pallet call, this direct assignment is possible.